src/main/resources/resource/WebGui/app/service/js/WebkitSpeechRecognitionGui.js

```diff
@@ -1,239 +1,290 @@
-            }
-            ;if (event.error == 'not-allowed') {
-                if (event.timeStamp - start_timestamp < 100) {
-                    $scope.status = 'Permission to use microphone is blocked. To change, go to chrome://settings/contentExceptions#media-stream';
-                    $scope.$apply();
+
+    $scope.setState = function(statusKey, event) {
+        console.debug('status ' + statusKey)
+
+        $scope.service.status = statusKey
+
+        switch (statusKey) {
+        case 'onstart':
+            $scope.service.isListening = true
+            $scope.service.img = '../WebkitSpeechRecognition/mic-animate.gif'
+            // $scope.errorText = null
+            $scope.startTimestamp = new Date().getTime()
+            console.debug('speak now')
+            $scope.$apply()
+            break
+
+        case 'onresult':
+            $scope.interimTranscript = ''
+
+            console.debug('onresult has ' + event.results.length + ' results')
+            for (var i = event.resultIndex; i < event.results.length; ++i) {
+                let data = event.results[i][0]
+                
+                $scope.recognizedResult = {
+                    text: data.transcript,
+                    confidence: (Math.round(data.confidence * 100) / 100).toFixed(2),
+                    isFinal: false
+                }
+
+                if (event.results[i].isFinal) {
+                    $scope.recognizedResult.isFinal = true
+                    msg.send('processResults', [$scope.recognizedResult])
+                    $scope.log.unshift($scope.recognizedResult)
+                    $scope.$apply()
                 } else {
-                    $scope.status = 'Permission to use microphone was denied.';
-                    $scope.$apply();
+                    // weird handling of this ... FIXME - just pubish it all and set the correct final ..
+                    $scope.interimTranscript += data.transcript
                 }
-                ;ignore_onend = true;
             }
-            ;
-        }
-        ;
-
-        // called when $scope.recognition finishes.
-        $scope.recognition.onend = function() {
-            mrl.sendTo($scope.service.name, "stopListening");
-            $scope.recognizing = false;
-            $scope.$apply();
-            if (ignore_onend) {
-                return;
+            break
+
+        case 'onend':
+            $scope.service.isListening = false
+            $scope.service.img = '../WebkitSpeechRecognition/mic-slash.png'
+            if (!$scope.stopRequested) {
+                $scope.restartCnt += 1
+                recognizer.start()
+            }
+            break
+
+        case 'onerror':
+            $scope.errorText = event.error
+            let errorTs = new Date().getTime()
+            if ((errorTs - $scope.startTimestamp) < 100) {
+                $scope.errorText += ' - high error rate - check other tabs for an active webkit speech recognizer, and close it'
```

Comment:
This error message is basically useless to people.  it is very misleading.  the old error messages should be restored.

Reply:
This needs work with the error state handling, which is not done yet.
I think having the "last" opened recognizer work is nice - vs, scouring through your tabs looking for the other one.  Its either that or the both get turned off.  The speech recognizer is global in chrome - only one tab can own the resource ... that's just the way it is ...